code/game/objects/items/devices/aicard.dm

```diff
@@ -24,29 +23,36 @@
 
 /obj/item/aicard/proc/update_state()
 	var/mob/living/silicon/ai/AI = locate(/mob/living/silicon/ai) in src //AI is inside.
+	update_icon(UPDATE_OVERLAYS)
 	if(AI)
 		name = "intelliCard - [AI.name]"
-		if(AI.stat == DEAD)
-			icon_state = "aicard-404"
-		else
-			icon_state = "aicard-full"
 		AI.cancel_camera() //AI are forced to move when transferred, so do this whenver one is downloaded.
 	else
-		icon_state = "aicard"
 		name = "intelliCard"
-		overlays.Cut()
+
+/obj/item/aicard/update_overlays()
+	. = ..()
+	var/mob/living/silicon/ai/AI = locate(/mob/living/silicon/ai) in src //AI is inside.
+	if(AI)
+		var/list/aicard_icon_state_names = icon_states(icon)
+		var/aicard_new_display = AI.icon_state
+
+		if(aicard_new_display in aicard_icon_state_names)
+			. += aicard_new_display
+		else if(AI.stat == DEAD)
```

Comment:
since this is an else if statement, doesn't this mean that even if the AI in the card is dead, it will still show the custom display instead of the `ai_dead` state?

Reply:
Yes, keep in mind that the two latter cases are backups in the case that an icon state matching that of the ais cannot be found. Under proper conditions only the first case will run, and this case works regardless of the condition of the ai

Reply:
There are alot of displays that have unique dead icon states, the first case uses them. The if dead check is for the case in which an icon state in the intelicard dmi cannot be found that matches the ais current disply.